expenses/views.py

```diff
@@ -67,6 +67,7 @@ def create(request):
             return redirect(reverse("expenses:detail", args=(o.id,)))
     else:
         form = ExpenseForm()
+        form.fields['category'].queryset = models.Category.objects.filter(user=request.user)
```

Comment:
refactor those two lines into `def get_expense_form(user, ....)` and use it in `update()` view as well.